plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/security/KafkaSslConfig.java

```diff
@@ -0,0 +1,174 @@
+    @ConfigDescription("The location of the trust store file")
+    public KafkaSslConfig setTruststoreLocation(String truststoreLocation)
+    {
+        this.truststoreLocation = truststoreLocation;
+        return this;
+    }
+
+    public Optional<String> getTruststorePassword()
+    {
+        return Optional.ofNullable(truststorePassword);
+    }
+
+    @Config("kafka.ssl.truststore.password")
+    @ConfigDescription("The password for the trust store file")
+    @ConfigSecuritySensitive
+    public KafkaSslConfig setTruststorePassword(String truststorePassword)
+    {
+        this.truststorePassword = truststorePassword;
+        return this;
+    }
+
+    public Optional<KafkaKeystoreTruststoreType> getTruststoreType()
+    {
+        return Optional.ofNullable(truststoreType);
+    }
+
+    @Config("kafka.ssl.truststore.type")
+    @ConfigDescription("The file format of the trust store file")
+    public KafkaSslConfig setTruststoreType(KafkaKeystoreTruststoreType truststoreType)
+    {
+        this.truststoreType = truststoreType;
+        return this;
+    }
+
+    public Optional<KafkaEndpointIdentificationAlgorithm> getEndpointIdentificationAlgorithm()
+    {
+        return Optional.ofNullable(endpointIdentificationAlgorithm);
+    }
+
+    @Config("kafka.ssl.endpoint-identification-algorithm")
+    @ConfigDescription("The endpoint identification algorithm to validate server hostname using server certificate")
+    public KafkaSslConfig setEndpointIdentificationAlgorithm(KafkaEndpointIdentificationAlgorithm endpointIdentificationAlgorithm)
+    {
+        this.endpointIdentificationAlgorithm = endpointIdentificationAlgorithm;
+        return this;
+    }
+
+    public Map<String, Object> getKafkaClientProperties()
+    {
+        ImmutableMap.Builder<String, Object> properties = ImmutableMap.builder();
+        addNotNull(properties, SslConfigs.SSL_KEYSTORE_LOCATION_CONFIG, keystoreLocation);
+        addNotNull(properties, SslConfigs.SSL_KEYSTORE_PASSWORD_CONFIG, keystorePassword);
+        addNotNull(properties, SslConfigs.SSL_KEYSTORE_TYPE_CONFIG, keystoreType);
+        addNotNull(properties, SslConfigs.SSL_KEY_PASSWORD_CONFIG, keyPassword);
+        addNotNull(properties, SslConfigs.SSL_TRUSTSTORE_LOCATION_CONFIG, truststoreLocation);
+        addNotNull(properties, SslConfigs.SSL_TRUSTSTORE_PASSWORD_CONFIG, truststorePassword);
+        addNotNull(properties, SslConfigs.SSL_TRUSTSTORE_TYPE_CONFIG, truststoreType);
+
+        if (getEndpointIdentificationAlgorithm().isPresent()) {
+            addNotNull(properties, SslConfigs.SSL_ENDPOINT_IDENTIFICATION_ALGORITHM_CONFIG, endpointIdentificationAlgorithm.getName());
+        }
+        else {
+            addNotNull(properties, SslConfigs.SSL_ENDPOINT_IDENTIFICATION_ALGORITHM_CONFIG, endpointIdentificationAlgorithm);
+        }
+
+        return properties.build();
+    }
+
+    private void addNotNull(ImmutableMap.Builder<String, Object> properties, String key, Object value)
```

Comment:
Use getters and `Optional::ifPresent`

Reply:
`getKeystoreLocation().ifPresent(v -> properties.add(SslConfigs.SSL_KEYSTORE_LOCATION_CONFIG, v);`